File: Codeforces/1520/C.kt
fun main() {
    val T = readLine()?.toInt() ?: 1
    for (t in 1..T) {
        val n = readLine()?.toInt() ?: 1
        when (n) {
            1 -> println(1)
            2 -> println(-1)
            else -> {
                val ans = Array(n) { IntArray(n) }
                var (value, flag) = arrayOf(1, 0)
                for (i in 0 until n) {
                    var j = flag
                    while (j < n) {
                        ans[i][j] = value++
                        j += 2
                    }
                    flag = flag.xor(1)
                }
                flag = 1
                for (i in 0 until n) {
                    var j = flag
                    while (j < n) {
                        ans[i][j] = value++
                        j += 2
                    }
                    flag = flag.xor(1)
                }
                ans.forEach { x ->
                    x.forEach { y ->
                        print(y)
                        print(" ")
                    }
                    println()
                }
            }
        }
    }
}

File: Codeforces/1520/C.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t, n;
    cin >> t;
    while (t--) {
        cin >> n;
        if (n == 1) {
            cout << "1\n";
        } else if (n == 2) {
            cout << "-1\n";
        } else {
            vector<vector<int>> ans(n, vector<int>(n));
            int val = 1;
            bool flag = true;
            for (int i = 0; i < n; ++i) {
                for (int j = (flag ? 0 : 1); j < n; j += 2) {
                    ans[i][j] = val++;
                }
                flag = !flag;
            }
            flag = true;
            for (int i = 0; i < n; ++i) {
                for (int j = (flag ? 1 : 0); j < n; j += 2) {
                    ans[i][j] = val++;
                }
                flag = !flag;
            }
            for (const auto &x : ans) {
                for (const auto &y : x) {
                    cout << y << " ";
                }
                cout << "\n";
            }
        }
    }
    return 0;
}